src/pool-bin/libraries/BinPool.sol

```diff
@@ -168,6 +169,14 @@ library BinPool {
                         amountsInWithFees = amountsInWithFees.sub(pFee);
                     }
 
+                    if (
+                        (binReserves.add(amountsInWithFees).sub(amountsOutOfBin)).getLiquidity(
```

Comment:
1/ i wonder if it will be cheaper in gas if you cast a variable to store `binReserves.add(amountsInWithFees).sub(amountsOutOfBin)` like how u did elsewhere 

2/ or if we apply the changes within `BinHelper.getAmountsOut` and `Binhelper.getAmountsIn` instead (as the price is already computed, we don't need to compute again like what we did here `swapState.activeId.getPriceFromId(params.binStep)`

Reply:
oh, that's actually what i did before, what it failed the forge coverage 😂

Reply:
But let me try option 2

Reply:
It did help on gas, thx 👍